tests/formats/lammps-atom.cpp

```diff
@@ -0,0 +1,553 @@
+    }
+
+    SECTION("Unrapped Coordinates") {
+        auto file = Trajectory("data/lammps/unwrapped.lammpstrj");
+        check_pos_representation(file);
+    }
+
+    SECTION("Scaled Unrapped Coordinates") {
+        auto file = Trajectory("data/lammps/scaled_unwrapped.lammpstrj");
+        check_pos_representation(file);
+    }
+
+    SECTION("Position Representation") {
+        auto file = Trajectory("data/lammps/detect_best_pos_repr.lammpstrj");
+        CHECK(file.nsteps() == 5);
+
+        Frame frame = file.read();
+        CHECK(frame.size() == 854);
+        CHECK(frame.step() == 100000);
+        CHECK(approx_eq((*frame.get("time")).as_double(), 25e9, 1e-6));
+        auto positions = frame.positions();
+
+        CHECK(approx_eq(positions[679], Vector3D(1.47679, -25.2886, 2.38234), 1e-3));
+        CHECK(approx_eq(positions[764], Vector3D(-256.58, 117.368, 1.9654), 1e-3));
+
+        frame = file.read();
+        CHECK(frame.size() == 854);
+        CHECK(frame.step() == 101000);
+        CHECK(!frame.get("time"));
+        positions = frame.positions();
+
+        CHECK(frame.cell().shape() == UnitCell::ORTHORHOMBIC);
+        CHECK(approx_eq(frame.cell().lengths(), Vector3D(60.0, 60.0, 250.0), 1e-2));
+        CHECK(approx_eq(frame.cell().angles(), Vector3D(90.0, 90.0, 90.0), 1e-2));
+
+        CHECK(approx_eq(positions[683], Vector3D(-43.3683, 322.948, 208.063), 1e-3));
+        CHECK(approx_eq(positions[828], Vector3D(150.083, -135.113, 189.641), 1e-3));
+
+        frame = file.read();
+        CHECK(frame.size() == 856);
+        CHECK(frame.step() == 102000);
+
+        positions = frame.positions();
+        CHECK(approx_eq(positions[747], Vector3D(-158.317, 142.593, 2.11392), 1e-3));
+        CHECK(approx_eq(positions[799], Vector3D(224.784, -167.878, 39.3765), 1e-3));
+
+        frame = file.read();
+        CHECK(frame.size() == 856);
+        CHECK(frame.step() == 103000);
+
+        positions = frame.positions();
+        CHECK(approx_eq(positions[735], Vector3D(67.2657, 30.0627, 2.1141), 1e-3));
+        CHECK(approx_eq(positions[775], Vector3D(125.347, -82.3507, 46.611), 1e-3));
+
+        frame = file.read();
+        CHECK(frame.size() == 856);
+        CHECK(frame.step() == 104000);
+
+        positions = frame.positions();
+        CHECK(approx_eq(positions[652], Vector3D(-188.131, 96.0777, 196.23), 1e-3));
+        CHECK(approx_eq(positions[838], Vector3D(-33.6068, -50.5113, 209.306), 1e-3));
+
+        CHECK_THROWS_AS(file.read(), FileError);
+    }
+
+    SECTION("Errors") {
+        auto file = Trajectory("data/lammps/broken.lammpstrj");
+        std::string msg = "can not read box header in LAMMPS format: expected an ITEM entry in "
+                          "LAMMPS format, got 'DUMMY'";
+        CHECK_THROWS_WITH(file.read_step(0), msg);
+        msg =
+            "can not read box header in LAMMPS format: missing 'BOX BOUNDS' item in LAMMPS format";
+        CHECK_THROWS_WITH(file.read_step(1), msg);
```

Comment:
Not sure if it's intended or not, but using `file.read()` here does not work. The file cursor is at the position of the last error instead of the position of the next frame, this means that `file.read()` simply starts reading the next line. In contrast to that, `frame.read_step(1)` starts at the correct position for the next frame.
IMHO this is a bug (but not a serious one) because as a user one gets the impression that an read error should be recoverable because we throw an error (or return an result in Rust).

Reply:
Oh, that's interesting. This should be not too hard to fix, I'll open an issue.